src/main/groovy/com/deploygate/gradle/plugins/tasks/LoginTask.groovy

```diff
@@ -16,141 +18,151 @@ import java.util.concurrent.TimeUnit
 import java.util.concurrent.TimeoutException
 
 class LoginTask extends DefaultTask {
-    @Internal def port = 0
 
-    @Internal CountDownLatch latch
-    boolean saved
-    @Internal CliCredentialStore localCredential
+    @Internal
+    DeployGateExtension deployGateExtension
+
+    @Internal
+    CliCredentialStore credentialStore
+
+    @Internal
+    String onetimeKey
+
+    @Internal
+    CountDownLatch countDownLatch
 
     @TaskAction
     def setup() {
-        localCredential = new CliCredentialStore()
-        if (!hasCredential())
-            if (!setupCredential())
-                throw new RuntimeException('Failed to retrieve DeployGate credentials. Please try again or specify it in your build.gradle script.')
-
-        project.deploygate.appOwnerName =
-                [project.deploygate.appOwnerName, System.getenv(DeployGatePlugin.ENV_NAME_APP_OWNER_NAME), System.getenv(DeployGatePlugin.ENV_NAME_APP_OWNER_NAME_V1), localCredential.name].find {
-                    it != null
-                }
-        project.deploygate.apiToken =
-                [project.deploygate.apiToken, System.getenv(DeployGatePlugin.ENV_NAME_API_TOKEN), localCredential.token].find {
-                    it != null
-                }
-    }
+        if (!(deployGateExtension.appOwnerName && deployGateExtension.apiToken)) {
+            if (!setupCredential()) {
+                throw new RuntimeException('We could not retrieve DeployGate credentials. Please make sure you have configured app owner name and api token or any browser application is available to launch the authentication flow.')
+            }
 
-    boolean hasCredential() {
-        hasCreadentialInScript() || hasSavedCredential() || hasCredentialInEnv()
-    }
+            println "Welcome ${credentialStore.name}!"
 
-    boolean hasCredentialInEnv() {
-        [System.getenv(DeployGatePlugin.ENV_NAME_APP_OWNER_NAME), System.getenv(DeployGatePlugin.ENV_NAME_APP_OWNER_NAME_V1)].any() &&
-                [System.getenv(DeployGatePlugin.ENV_NAME_API_TOKEN)].any()
-    }
+            logger.info("The authentication has succeeded. The application owner name is ${credentialStore.name}.")
 
-    // From Gradle 7.0, only one method to get boolean property value is allowed
-    // whereas Groovy generates both isSaved() and getSaved(), Gradle 7.0 considers 2 getters exist for saved property.
-    // To suppress getSaved() we explicitly define isSaved().
-    private boolean isSaved() {
-        saved
-    }
+            // We can set the values iff it's not set yet because of the idempotency.
 
-    private boolean hasSavedCredential() {
-        localCredential.name && localCredential.token
-    }
+            if (!deployGateExtension.appOwnerName) {
```

Comment:
The previous code sets the value by looking up their priority though, the current code has already done this. So it's okay to just set the credentials iff missing.

```
project.deploygate.appOwnerName =
                [project.deploygate.appOwnerName, System.getenv(DeployGatePlugin.ENV_NAME_APP_OWNER_NAME), System.getenv(DeployGatePlugin.ENV_NAME_APP_OWNER_NAME_V1), localCredential.name].find {
                    it != null
                }
        project.deploygate.apiToken =
                [project.deploygate.apiToken, System.getenv(DeployGatePlugin.ENV_NAME_API_TOKEN), localCredential.token].find {
                    it != null
                }
```